xdsl/dialects/smt.py

```diff
@@ -308,5 +377,5 @@ def __init__(self, *values: SSAValue):
         ForallOp,
         YieldOp,
     ],
-    [BoolType],
+    [BoolType, FuncType],
```

Comment:
```suggestion
    [BoolType, FuncType,],
```
adding this before @superlopuh does